packages/javascript/bh-shared-ui/src/components/UpdateUserForm/UpdateUserForm.tsx

```diff
@@ -56,427 +69,681 @@ const UpdateUserForm: React.FC<{
+                first_name: getUserQuery.data.first_name || '',
+                last_name: getUserQuery.data.last_name || '',
+                sso_provider_id: getUserQuery.data.sso_provider_id?.toString() || undefined,
+                roles: getUserQuery.data.roles ? getUserQuery.data.roles?.map((role: any) => role.id) : [],
+                all_environments: getUserQuery.data.all_environments,
+                environment_targeted_access_control: {
+                    environments:
+                        getUserQuery.data.all_environments === false
+                            ? getUserQuery.data.environment_targeted_access_control?.map(
+                                  (environment: EnvironmentRequest) => environment
+                              )
+                            : null,
+                },
             }}
-            roles={getListDisplayRolesQuery.data}
-            SSOProviders={listSSOProvidersQuery.data}
+            error={error}
             hasSelectedSelf={hasSelectedSelf}
             isLoading={isLoading}
-            error={error}
+            onSubmit={onSubmit}
+            roles={getRolesQuery.data}
+            SSOProviders={listSSOProvidersQuery.data}
+            showEnvironmentAccessControls={showEnvironmentAccessControls}
         />
     );
 };
 
 const UpdateUserFormInner: React.FC<{
-    onCancel: () => void;
-    onSubmit: (user: UpdateUserRequestForm) => void;
+    error: any;
+    hasSelectedSelf: boolean;
     initialData: UpdateUserRequestForm;
+    isLoading: boolean;
+    onSubmit: (user: UpdateUserRequestForm) => void;
     roles?: Role[];
+    showEnvironmentAccessControls?: boolean;
     SSOProviders?: SSOProvider[];
-    hasSelectedSelf: boolean;
-    isLoading: boolean;
-    error: any;
-}> = ({ onCancel, onSubmit, initialData, roles, SSOProviders, hasSelectedSelf, isLoading, error }) => {
-    const {
-        control,
-        handleSubmit,
-        setValue,
-        formState: { errors },
-        setError,
-        watch,
-    } = useForm<UpdateUserRequestForm & { authenticationMethod: 'sso' | 'password' }>({
+}> = ({
+    error,
+    hasSelectedSelf,
+    initialData,
+    isLoading,
+    onSubmit,
+    roles,
+    showEnvironmentAccessControls,
+    SSOProviders,
+}) => {
+    const form = useForm<UpdateUserRequestForm & { authenticationMethod: 'sso' | 'password' }>({
         defaultValues: {
             ...initialData,
-            authenticationMethod: initialData.SSOProviderId ? 'sso' : 'password',
+            authenticationMethod: initialData.sso_provider_id ? 'sso' : 'password',
         },
     });
 
-    const authenticationMethod = watch('authenticationMethod');
+    const [selectedRoleValue, setSelectedRoleValue] = useState<number[]>(initialData.roles);
+    const [searchInput, setSearchInput] = useState<string>('');
+    const roleInputValue = form.watch('roles');
+    const selectedRole = roleInputValue.toString() === '2' || roleInputValue.toString() === '3';
```

Comment:
Looks like this same evaluation is done below for the selectedETACEnabledRole variable, do we need both?